app/Http/Controllers/Crew/EditionController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers\Crew;
+
+use App\Http\Controllers\Controller;
+use App\Models\Edition;
+use App\Models\Event;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\View\View;
+
+class EditionController extends Controller
+{
+    /**
+     * Returns the landing page
+     * @return View
+     */
+    public function index() : View
+    {
+        $editions = Edition::all();
+
+        return view('moderator.editions.index', compact('editions'));
+    }
+
+    /**
+     * Returns create form for the editions
+     * @return View
+     */
+    public function create() : View
+    {
+        return view('moderator.editions.create');
+    }
+
+    /**
+     * Creates new instance of Edition and stores in the database
+     * @param Request $request data about the edition
+     * @return RedirectResponse redirects to show page
+     */
+    public function store(Request $request) : RedirectResponse
+    {
+        $edition = Edition::create($request->validate(Edition::rules()));
+
+        // attach all the default events to a new edition
+        foreach (Event::all() as $event) {
+            $edition->addEvent($event);
+        }
+
+        return redirect(route('moderator.events.index', compact('edition')))
+            ->banner("Edition {$edition->name} was successfully created");
+    }
+
+    /**
+     * Triggers a function for a specific edition that activates it
+     * @param Edition $edition to activate
+     * @return RedirectResponse redirects to the index page
+     */
+    public function activateEdition(Edition $edition) : RedirectResponse
+    {
+        // if all the dates for edition are present, it can be activated
+        if ($edition->configured()) {
+            $edition->activate();
+        }
+
+        return redirect(route('moderator.editions.index'))
+            ->banner("Edition {$edition->name} was successfully activated");
+    }
```

Comment:
![image](https://github.com/HZ-HBO-ICT/it-conference/assets/77020646/0507ef36-44e2-4f02-85cd-8aa2a16c1883)

This happens when I try to activate an edition, therefore I couldn't test the rest.

Reply:
That's very interesting!

Reply:
Fixed it, in the modal it was still checking for the old role `content moderator` instead of `event organizer`. Also, this issue exists in the `delete-presentation` modal, though the functionality magically works...